src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,114 @@
+package duke.storage;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+
+import duke.exception.DukeException;
+import duke.exception.ErrorString;
+import duke.task.DeadLine;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * Storage handles the updating and loading of tasks with the specified file path.
+ *
+ * @author yl-ang
+ */
+public class Storage {
+    private String filePath;
+
+    /**
+     * Constructs a Storage object with the specified filePath.
+     *
+     * @param filePath FilePath the path to the file containing the tasks.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
```

Comment:
The code looks incredibly good :thumbsup: . I just have a question: does this handle the file path on different OS?

Reply:
Thank you for the advice, I will test it on other OS (such as Mac) and check for consistency 👍 😄